### ISSUE TITLE
NextFocusApi not working for inside Animated View or AnimatedImageBackground

### ISSUE THREAD

**Issue by aniketgupta8960** [react-native-tvos/react-native-tvos#383]
### Description

HI Team,
I am working on a react native game project based on tvOs where there are animation of button.The next focus api works fine with normal view but inside the animated view the next focus api doesn't work.
Please help me with this issue as this is a blocker for us

### Version

0.68.2-2

### Output of `npx react-native info`

System:
    OS: macOS 12.6
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 27.93 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.0.0 - /usr/local/bin/node
    Yarn: 1.22.10 - ~/‘~/.npm-global/bin/yarn
    npm: 8.5.5 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    Android SDK:
      API Levels: 21, 23, 25, 27, 28, 29, 30, 31, 33
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0
      System Images: android-28 | Google Play Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google TV Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: 22.1.7171670
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6953283
    Xcode: 14.0.1/14A400 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.12 - /usr/local/opt/openjdk@11/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: Not Found
    react-native-macos: Not Found
    react-native-tvos:  0.68.2-2 
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Put buttons inside Animated.View  and try to use the nextFocus api for button that are not in grid then the next focus doesn't work. 
The code below works fine with ImageBackground but as I have created a new AnimatedImageBackground using     
const AnimatedImageBackground = useRef(Animated.createAnimatedComponent(ImageBackground)).current;
Therefore nextFocus Api is not working in this

### Snack, code example, screenshot, or link to a repository

  <View style={styles.container1}>
            <AnimatedImageBackground source={highlight_frame}
                style={[{
                    left: deviceDimensions.width * 0.5 - scales(1590) * 0.5,
                    top: scales(43),
                    width: scales(1590),
                    height: scales(1450),
                    position: 'absolute',
                }, props.animatedStyle]}>
                <View style={{
                    marginTop: scales(65),
                    marginLeft: scales(400),
                    alignItems: 'center',
                    width: scales(800),
                }}>
                    <Text
                        adjustsFontSizeToFit={true}
                        numberOfLines={1}
                        style={{
                            fontSize: fontSizeVertical(24),
                            fontFamily: fonts.stcBold,
                            color: 'white',
                            //textAlign: 'center',
                        }}> {strings.REPEAT_POPUP_HEADING} </Text>

                </View>

                <View style={{ flexDirection: strings.getLanguage() != 'ar' ? 'row-reverse' : 'row', justifyContent: 'center', position: 'absolute', top: scales(280), left: scales(150) }}>
                    <View style={{ alignItems: 'center' }}>
                        <View
                            style={{
                                width: scales(305),
                                height: scales(310),
                                justifyContent: 'center',
                                alignItems: 'center',
                            }}>
                            <Button
                                image={RepeatButton4}
                                hasTVPreferredFocus={currentActiveButton === 4}
                                imageOnPress={RepeatButton4Pressed}
                                textStyle={{ textAlign: 'center', fontSize: fontSizeVertical(40), fontFamily: fonts.stcBold, color: 'white' }}
                                buttonText1={repeatObject.repeatCount <= 3 ? '4' : repeatObject.repeatCount?.toString()}
                                textViewStyle={{ justifyContent: 'center', alignItems: 'center' }}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    if (repeatObject.repeatCount <= 3) {
                                        setRepeatObject((prev) => {
                                            return { ...prev, repeatCount: 4 }
                                        })
                                    }
                                }}
                            />
                        </View>
                        <View >
                            <Image
                                style={{
                                    width: scales(107),
                                    height: scales(113),
                                    alignItems: 'center',
                                }}
                                source={repeatObject.repeatCount > 3 ? RadioButtonPressed : RadioButton}
                            />
                        </View>
                        <View
                            style={{
                                top: scales(90),
                                left: scales(-30),
                                width: scales(109),
                                height: scales(119),
                                position: 'absolute',
                                transform: [{ rotate: '180deg' }]
                            }}>
                            <Button
                                hasTVPreferredFocus={currentActiveButton === 5}
                                ref={(component) => setDecrementReference(component)}
                                nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(thirdReference) : tagForDestination(decrementReference)}
                                nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(incrementReference) : tagForDestination(incrementReference)}
                                nextFocusDown={tagForDestination(fromReference)}
                                image={RepeatDecrement}
                                imageOnPress={RepeatDecrementPressed}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    if (repeatObject.repeatCount > 4) {
                                        setRepeatObject((prev) => {
                                            return { ...prev, repeatCount: prev.repeatCount - 1 }
                                        })
                                    }
                                }}
                            />
                        </View>
                        <View
                            style={{
                                top: scales(90),
                                left: scales(220),
                                width: scales(109),
                                height: scales(119),
                                position: 'absolute',
                                transform: [{ rotate: '180deg' }]
                            }}>
                            <Button
                                hasTVPreferredFocus={currentActiveButton === 4}
                                ref={(component) => setincrementReference(component)}
                                nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(decrementReference) : tagForDestination(decrementReference)}
                                nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(incrementReference) : tagForDestination(thirdReference)}
                                nextFocusDown={tagForDestination(fromReference)}
                                image={RepeatIncrement}
                                imageOnPress={RepeatIncrementPressed}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    if (repeatObject.repeatCount <= 3) {
                                        setRepeatObject((prev) => {
                                            return { ...prev, repeatCount: 4 }
                                        })
                                    }
                                    if (repeatObject.repeatCount >= 4 && repeatObject.repeatCount < 20) {
                                        setRepeatObject((prev) => {
                                            return { ...prev, repeatCount: prev.repeatCount + 1 }
                                        })
                                    }
                                }}
                            />
                        </View>
                    </View>
                    <View style={{ alignItems: 'center', marginHorizontal: scales(15) }}>
                        <View
                            style={{
                                width: scales(305),
                                height: scales(310),
                                justifyContent: 'center',
                                alignItems: 'center'
                            }}>
                            <Button
                                hasTVPreferredFocus={currentActiveButton === 3}
                                ref={(component) => setThirdReference(component)}
                                nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(secondReference) : tagForDestination(incrementReference)}
                                nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(decrementReference) : tagForDestination(secondReference)}
                                nextFocusDown={tagForDestination(fromReference)}
                                image={RepeatButton3}
                                imageOnPress={RepeatButton3Pressed}
                                textStyle={{ textAlign: 'center', fontSize: fontSizeVertical(40), fontFamily: fonts.stcBold, color: 'white' }}
                                buttonText1={'3'}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    setRepeatObject((prev) => {
                                        return { ...prev, repeatCount: 3 }
                                    })
                                }}
                            />
                        </View>
                        <View>
                            <Image
                                style={{
                                    width: scales(107),
                                    height: scales(113),
                                }}
                                source={repeatObject.repeatCount == 3 ? RadioButtonPressed : RadioButton}
                            />
                        </View>
                    </View>
                    <View style={{ alignItems: 'center', marginHorizontal: scales(15) }}>
                        <View
                            style={{
                                width: scales(305),
                                height: scales(310),
                                justifyContent: 'center',
                                alignItems: 'center'
                            }}>
                            <Button
                                //hasTVPreferredFocus={currentActiveButton === 2}
                                ref={(component) => setSecondReference(component)}
                                nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(firstReference) : tagForDestination(thirdReference)}
                                nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(thirdReference) : tagForDestination(firstReference)}
                                nextFocusDown={tagForDestination(fromReference)}
                                image={RepeatButton2}
                                imageOnPress={RepeatButton2Pressed}
                                textStyle={{ textAlign: 'center', fontSize: fontSizeVertical(40), fontFamily: fonts.stcBold, color: 'white' }}
                                buttonText1={'2'}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    setRepeatObject((prev) => {
                                        return { ...prev, repeatCount: 2 }
                                    })
                                }}
                            />
                        </View>
                        <View>
                            <Image
                                style={{
                                    width: scales(107),
                                    height: scales(113),
                                }}
                                source={repeatObject.repeatCount == 2 ? RadioButtonPressed : RadioButton}
                            />
                        </View>
                    </View>
                    <View style={{ alignItems: 'center' }}>
                        <View
                            style={{
                                width: scales(305),
                                height: scales(310),
                                justifyContent: 'center',
                                alignItems: 'center'
                            }}>
                            <Button
                                image={RepeatButton1}
                                ref={(component) => setFirstReference(component)}
                                hasTVPreferredFocus={currentActiveButton === 1}
                                nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(firstReference) : tagForDestination(secondReference)}
                                nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(secondReference) : tagForDestination(firstReference)}
                                nextFocusDown={tagForDestination(fromReference)}
                                imageOnPress={RepeatButton1Pressed}
                                textStyle={{ textAlign: 'center', fontSize: fontSizeVertical(40), fontFamily: fonts.stcBold, color: 'white' }}
                                buttonText1={'1'}
                                onPress={() => {
                                    playAudio('all_clicks.mp3');
                                    setRepeatObject((prev) => {
                                        return { ...prev, repeatCount: 1 }
                                    })
                                }}
                            />
                        </View>
                        <View>
                            <Image
                                style={{
                                    width: scales(107),
                                    height: scales(113),
                                }}
                                source={repeatObject.repeatCount <= 1 ? RadioButtonPressed : RadioButton}
                            />
                        </View>
                    </View>
                </View>
                <View style={{ justifyContent: 'flex-end', flexDirection: 'row', alignItems: 'center', position: 'absolute', bottom: scales(558), right: strings.getLanguage() != 'ar' ? scales(220) : scales(260) }}>
                    <View >
                        <View style={{ flexDirection: strings.getLanguage() != 'ar' ? 'row-reverse' : 'row', alignItems: 'center' }}>
                            <Text style={{ fontSize: fontSizeVertical(29), fontFamily: fonts.stcBold, color: 'white' }}>{strings.REPEAT_ONCE}</Text>
                            <Image
                                style={{
                                    width: scales(160),
                                    height: scales(160),
                                    marginHorizontal: scales(15)
                                }}
                                source={RepeatLoop}
                            />
                            <Image
                                style={{
                                    width: scales(107),
                                    height: scales(113),
                                }}
                                source={repeatObject.rangeSelect ? RadioButton : RadioButtonPressed}
                            />
                        </View>
                    </View>
                </View>
                <View style={{ justifyContent: 'flex-end', flexDirection: strings.getLanguage() != 'ar' ? 'row-reverse' : 'row', alignItems: 'center', position: 'absolute', bottom: scales(390), left: strings.getLanguage() != 'ar' ? scales(250) : scales(228) }}>
                    <Text style={{ fontSize: fontSizeVertical(29), fontFamily: fonts.stcBold, color: 'white' }}>{strings.REPEAT_RANGE}</Text>
                    <ImageBackground
                        style={{
                            width: scales(299),
                            height: scales(130),
                            marginHorizontal: scales(15),
                        }}
                        source={RepeatDropDown}>
                        <TextPressableComponent
                            onPress={() => {
                                setIsToDropDownVisible(true)
                                setCurrentActiveButton(2);
                            }}
                            ref={(component) => setToReference(component)}

                            nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(toReference) : tagForDestination(fromReference)}
                            nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(fromReference) : tagForDestination(toReference)}
                            nextFocusUp={tagForDestination(secondReference)}
                            nextFocusDown={tagForDestination(submitReference)}
                            hasTVPreferredFocus={currentActiveButton === 7}
                            style={{ height: '100%', zIndex: 20 }}
                            normalView={<View style={{ alignItems: 'center', justifyContent: 'center', height: '100%', paddingBottom: scales(30), paddingRight: scales(100) }}>
                                <Text style={{ fontSize: fontSizeVertical(18), fontFamily: fonts.stcBold, color: '#A768C9' }}>{repeatObject.rangeSelect ? (repeatObject.to * 1 - (1 - props.exceptionWordNoSubstract)).toString() : strings.REPEAT_TO}</Text>
                            </View>}
                            highlightedView={<View style={{ alignItems: 'center', justifyContent: 'center', height: '100%', paddingBottom: scales(30), paddingRight: scales(100) }}>
                                <Text style={{ fontSize: fontSizeVertical(18), fontFamily: fonts.stcBold, color: 'yellow' }}>{repeatObject.rangeSelect ? (repeatObject.to * 1 - (1 - props.exceptionWordNoSubstract)).toString() : strings.REPEAT_TO}</Text>
                            </View>}
                        />
                        {isToDropDownVisible && <View
                            style={{
                                position: 'absolute',
                                zIndex: 50,
                                width: scales(280),
                                bottom: scales(20),
                                backgroundColor: '#440971',
                                borderRadius: scales(40),
                                left: scales(5)
                            }}
                        >
                            <FlatList
                                maxHeight={scales(700)}
                                data={toDropDownValues}
                                renderItem={({ item, index }) => {
                                    return (
                                        <View>
                                            <Pressable
                                                onPress={(selectedItem) => {
                                                    setRepeatObject((prev) => {
                                                        return {
                                                            ...prev,
                                                            to: (item * 1 + (1 - props.exceptionWordNoSubstract)).toString(),
                                                            rangeSelect: true
                                                        }
                                                    })

                                                    setIsToDropDownVisible(false)
                                                }}

                                                hasTVPreferredFocus={index == 0 && currentActiveButton === 2}
                                            >
                                                {({ pressed, focused }) => (
                                                    <>
                                                        {pressed || focused ? (
                                                            <View style={{ backgroundColor: '#9700FF', borderRadius: scales(40) }}>
                                                                <Text
                                                                    style={styles.displayOptionText}
                                                                >
                                                                    {item}
                                                                </Text>
                                                            </View>

                                                        ) : (
                                                            <View>
                                                                <Text
                                                                    style={styles.displayOptionText}
                                                                >
                                                                    {item}
                                                                </Text>

                                                            </View>

                                                        )}
                                                    </>
                                                )}
                                            </Pressable>

                                        </View>
                                    );
                                }}
                            />
                        </View>}
                    </ImageBackground>
                    <ImageBackground
                        style={{
                            width: scales(299),
                            height: scales(130),
                            marginHorizontal: scales(15)
                        }}
                        source={RepeatDropDown}>
                        <TextPressableComponent
                            hasTVPreferredFocus={currentActiveButton === 6}
                            ref={(component) => setFromReference(component)}
                            nextFocusRight={strings.getLanguage() != 'ar' ? tagForDestination(toReference) : tagForDestination(fromReference)}
                            nextFocusLeft={strings.getLanguage() != 'ar' ? tagForDestination(fromReference) : tagForDestination(toReference)}
                            nextFocusUp={tagForDestination(secondReference)}
                            nextFocusDown={tagForDestination(submitReference)}
                            onPress={() => {
                                setIsFromDropDownVisible(true)
                            }}
                            style={{ height: '100%', zIndex: 20 }}
                            normalView={<View style={{ alignItems: 'center', justifyContent: 'center', height: '100%', paddingBottom: scales(30), paddingRight: scales(100) }}>
                                <Text style={{ fontSize: fontSizeVertical(18), fontFamily: fonts.stcBold, color: '#A768C9' }}>{repeatObject.rangeSelect ? (repeatObject.from * 1 - (1 - props.exceptionWordNoSubstract)).toString() : strings.REPEAT_FROM}</Text>
                            </View>}
                            highlightedView={<View style={{ alignItems: 'center', justifyContent: 'center', height: '100%', paddingBottom: scales(30), paddingRight: scales(100) }}>
                                <Text style={{ fontSize: fontSizeVertical(18), fontFamily: fonts.stcBold, color: 'yellow' }}>{repeatObject.rangeSelect ? (repeatObject.from * 1 - (1 - props.exceptionWordNoSubstract)).toString() : strings.REPEAT_FROM}</Text>
                            </View>}
                        />
                        {isFromDropDownVisible && <View
                            style={{
                                position: 'absolute',
                                zIndex: 50,
                                // height: scales(1200),
                                width: scales(280),
                                bottom: scales(20),
                                backgroundColor: '#440971',
                                borderRadius: scales(40),
                                left: scales(5)
                            }}
                        >
                            <FlatList
                                maxHeight={scales(700)}
                                data={fromDropDownValues}
                                renderItem={({ item, index }) => {
                                    return (
                                        <View>
                                            <Pressable
                                                onPress={(selectedItem) => {
                                                    setRepeatObject((prev) => {
                                                        return { ...prev, from: (item * 1 + (1 - props.exceptionWordNoSubstract)).toString(), to: (item * 1 + (1 - props.exceptionWordNoSubstract)).toString(), rangeSelect: true }
                                                    });
                                                    setIsFromDropDownVisible(false)
                                                    //toDropdownRef.current.reset()
                                                }}

                                                hasTVPreferredFocus={currentActiveButton === index}
                                            >
                                                {({ pressed, focused }) => (
                                                    <>
                                                        {pressed || focused ? (
                                                            <View style={{ backgroundColor: '#9700FF', borderRadius: scales(40) }}>
                                                                <Text
                                                                    style={styles.displayOptionText}
                                                                >
                                                                    {item}
                                                                </Text>
                                                            </View>

                                                        ) : (
                                                            <View>
                                                                <Text
                                                                    style={styles.displayOptionText}
                                                                >
                                                                    {item}
                                                                </Text>

                                                            </View>

                                                        )}
                                                    </>
                                                )}
                                            </Pressable>
                                        </View>
                                    );
                                }}
                            />
                        </View>}

                    </ImageBackground>
                    <View>
                        <Image
                            style={{
                                width: scales(107),
                                height: scales(113),
                            }}
                            source={repeatObject.rangeSelect ? RadioButtonPressed : RadioButton}
                        />
                    </View>
                </View>
                <View
                    style={{
                        top: scales(1063),
                        left: scales(1590) * 0.5 - scales(545) * 0.5,
                        width: scales(545),
                        height: scales(286),
                        position: 'absolute',
                    }}>
                    <Button
                        image={RepeatOkButton}
                        ref={(component) => setSubmitReference(component)}
                        nextFocusUp={tagForDestination(toReference)}
                        hasTVPreferredFocus={currentActiveButton === 8}
                        imageOnPress={RepeatOkButtonPressed}
                        textStyle={{ fontSize: fontSizeVertical(29), fontFamily: fonts.stcBold, color: 'white' }}
                        buttonText1={strings.REPEAT_SUBMIT}
                        imageStyle={{ alignItems: 'center', justifyContent: 'flex-start', paddingTop: scales(10) }}
                        imageStyleOnPRess={{ alignItems: 'center', justifyContent: 'flex-start', paddingTop: scales(10) }}
                        onPress={() => {
                            playAudio('all_clicks.mp3');
                            props.onSubmit(repeatObject)
                        }}
                    />
                </View>
            </AnimatedImageBackground>
        </View>

**Comment by douglowder** [comment]
Please provide a complete project that shows the issue, and I will be happy to help investigate.

**Comment by aniketgupta8960** [comment]
HI @douglowder  This is the drive link for the code https://drive.google.com/file/d/1odCAl7FB5dkCu1jGn92flpL_O-7nmtU-/view?usp=sharing

**Comment by douglowder** [comment]
Thanks @aniketgupta8960 -- unfortunately, that archive will not build, due to missing assets.

It would be helpful to make a much simpler app that only has one screen that shows the problem. You can also keep the file size much much smaller by not including the `node_modules` directory or the `ios/Pods` directory in the archive.

**Comment by aniketgupta8960** [comment]
Hi @douglowder Sorry to hear that the build didn't run on your side but now I have removed all the missing assets and please if you can download the code from this link https://drive.google.com/file/d/1odCAl7FB5dkCu1jGn92flpL_O-7nmtU-/view?usp=sharing
and test the navigation on Repeat Popup which appears on clicking the rightMost button on the screen .

**Comment by douglowder** [comment]
@aniketgupta8960 good news -- I was able to run your app and see the issue.

It isn't clear exactly what the problem is. I was able to show that in some cases, focus can move to buttons in the screen underneath the RepeatPopup modal. Unfortunately, I don't think I'm going to have an answer for you very quickly.

I'll have to see if I can simplify the repro. With all the many buttons all having preferred focus under different states, it's very complex to debug with this app.
